app/views/layouts/_nav_bar.html.erb

```diff
@@ -25,7 +25,7 @@
         <% end %>
       <% end %>
     <% else %>
-      <%= menu_item "Welcome, #{current_user.first_name}" %>
+      <%= menu_item "Welcome, #{current_user.username}" %>
```

Comment:
I think this should be an or statement since currently users are not *required* to have a username.